microsoft-edge/devtools-guide-chromium/evaluate-performance/index.md

```diff
@@ -183,22 +176,58 @@ This article gives you a lot to learn. But now you have a solid foundation in th
 
 Using the workflows and tools that you just learned, click **Optimize** on the demo to turn on the optimized code, take another performance recording, and then analyze the results.  From the improved framerate to the reduction in events in the flame chart in the **Main** section, the optimized version of the app does much less work, resulting in better performance.
 
-The optimized code uses a different sequence of actions to do less work. Note that this code could be made even faster by only using properties that only affect compositing, instead of manipulating the `top` property of every icon.
-<!--  > For more information, see [Use transform and opacity changes for animations](https://web.dev/stick-to-compositor-only-properties-and-manage-layer-count/#use-transform-and-opacity-changes-for-animations). todo: add rendering section when available -->
-
+Compare this snippet of JavaScript from the unoptimized version of the app:
+
+```javascript
+var pos = m.classList.contains("down")
+  ? m.offsetTop + distance
+  : m.offsetTop - distance;
+if (pos < 0) pos = 0;
+if (pos > maxHeight) pos = maxHeight;
+m.style.top = pos + "px";
+if (m.offsetTop === 0) {
+  m.classList.remove("up");
+  m.classList.add("down");
+}
+if (m.offsetTop === maxHeight) {
+  m.classList.remove("down");
+  m.classList.add("up");
+}
+```
+
+`m` references each blue icon on the page. In the unoptimized version, we refer to `pos` as the new vertical position of the icon, calculated based on `m.offsetTop + distance` if the icon is moving down and where `distance` is a fixed increment.  If the icon is moving up instead, the new position is instead calculated with `m.offsetTop - distance`.
+
+We check if we've exceeded either `0` or `maxHeight` as the new position whether we're moving up or down, adjust accordingly, and then we finally set the new position for each icon in CSS with `m.style.top = pos + "px";`.
```

Comment:
```suggestion
The above code snippet runs on every frame of the browser rendering loop and for each blue icon on the page. The `m` variable references a single blue icon.

In this unoptimized version, we create a `pos` variable that's set to the current position of the icon to which we add some distance. The current position of the icon is read by using `m.offsetTop`.

After making sure that the icon is still within the bounds of the page, we set its new position by using `m.style.top` which sets inline styles on the element.

Finally, we read `m.offsetTop` again to adjust the direction of the icon.
```

I'm hoping this helps clarify the code snippet a bit more. I removed some of the up/down details since the really important parts here are how we read and write the position of the icon.

Reply:
I'm wondering if this is where I asked to pls add a direct statement that positively states what the optimization is/does.  I think my ask was re: this article.  Does this PR w/ this change, now have a simple, direct, positive summary of what the optimization is?

Reply:
i would add comma:
```
that's set to the current position of the icon, to which we add some distance.
```

Reply:
``
we set its new position by using `m.style.top`, which sets inline styles on the element.
```
I would add comma, per style guide.
https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/t/that-vs-which

Reply:
add comma:
```
read `m.offsetTop` again, to adjust the direction of the icon.
```